falcon/request.py

```diff
@@ -2066,6 +2077,16 @@ def get_param_as_list(
         if name in params:
             items = params[name]
 
+            # If a delimiter is specified AND the param is a single string, split it.
+            if delimiter and isinstance(items, str):
```

Comment:
Let's only allow the delimiters from the OpenAPI spec in the first iteration, so `','`, `' '`, and `'|'`.
This will make this change easier to reason about in the terms of possible edge cases and side effects.

Reply:
it may makes sense to use a dict like similar to the one below to validate delimiter
```py
{' ': ' ', 'spaceDelimited': ' ', ',': ',', 'commaDelimited': ',', '|': '|', 'pipeDelimited': '|'}
```

(we could likely define this at module level

Reply:
(not sure if we want to also support `comma, space, pipe` as valid keys. I'm not against it, but they are not in the spec)